### PR TITLE
changes for modifying tables

### PR DESCRIPTION
**Pull request by nv1-0**

### CODE REVIEW

**Comment by hitman99** [comment]
Tests are failing. I cannot merge this without tests working.

**Comment by nv1-0** [comment]
@hitman99 the test cases are written now and the pipeline is passed.

**Review by KristianLauttamus — APPROVED** [review]

**Comment by rvora** [comment]
Is there any chance this pull request will be merged soon? I am seeing the same issue taking-over existing databases with public schema. Thanks.

File: pkg/postgres/database.go [hunk 1]
```diff
@@ -34,6 +36,27 @@ func (c *pg) CreateDB(dbname, role string) error {
...
 }
 
+func (c *pg) ModifyTables(dbname, role string, logger logr.Logger) error {
+
+	tmpDb := GetConnection(c.user, c.pass, c.host, dbname, c.args, logger)
+	defer tmpDb.Close()
+
+	_, err := tmpDb.Exec(CREATE_FUNCTION)
+	if err != nil {
+		_, err := tmpDb.Exec(fmt.Sprintf(ALTER_TABLE_OWNER, role))
```
Comment (arnarg) [inline pkg/postgres/database.go:46]: Please don't execute `ALTER_TABLE_OWNER` anyway. I guess you're doing this in the case where the function already exists in which case you should check for the correct error code like is done here: https://github.com/movetokube/postgres-operator/blob/master/pkg/postgres/role.go#L25

According to the docs duplicate function is `42723`.
https://www.postgresql.org/docs/current/errcodes-appendix.html

**Review by arnarg — COMMENTED** [review]
I'm not sure I like the idea of creating a generic execute function and using sql query execution is the best approach. I would rather query for tables with public schema and then loop over those in go and execute the alter table on them.

@hitman99 What is you opinion on this?

File: pkg/postgres/database.go (hunk 1 above)
Comment (hitman99) [reply pkg/postgres/database.go:46]: Agree with @arnarg generic execute is not a good idea and looks sketchy. Why do you need a generic function to execute an alter statement?

**Comment by hitman99** [comment]
Finally found time to look at this. I don't think that this is the right way to do it. This will mess up the permission system that we already have working. I'm not even sure this code should be present in the operator. The thing you want to do is a one-time thing, the ownership needs to be transferred once for the existing database. I think that this should be achieved in these steps:

1. Create an owner role
2. Transfer database and tables to this new owner role
3. Use newly created role in `spec.masterRole`
4. The operator can now manage the database without additional code (this PR)

What do you think @arnarg  ?

**Comment by arnarg** [comment]
I think that should be acceptable as like you said it's a one-time operation and this logic could complicate things. Especially as there are infinite ways permissions can be set currently and supporting a migration to the postgres-operator way of managing it would probably be difficult and nasty.

_Maybe_ we want to support some sort way to run arbitrary SQL in sort of hooks way (pre-create db, post-create db, ...) to support these custom migrations of databases into postgres-operator. This could be another CRD that is linked with it or something.

At least as the PR is now I will not accept it.

**Comment by hitman99** [comment]
> _Maybe_ we want to support some sort way to run arbitrary SQL in sort of hooks way (pre-create db, post-create db, ...) to support these custom migrations of databases into postgres-operator. This could be another CRD that is linked with it or something.
>
I think that is way better idea than current implementation in this PR.
> At least as the PR is now I will not accept it.
>
Agree 100%. I'm sorry @nv1-0, but I'll have to reject this PR as it does not fit in this code base as it exists now.

**Pull request closed without merging**